api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// Rule defines the single authorization rule.
+type Rule struct {
+	// Subjects contains the subject configuration.
+	// If empty, all subjects are included.
+	//
+	// +optional
+	Subjects []Subject `json:"subjects,omitempty"`
+
+	// Permissions contains allowed HTTP methods.
+	// If empty, all methods are matching.
+	//
+	// +optional
+	Permissions []string `json:"permissions,omitempty"`
+
+	// Action defines the action to be taken if the rule matches.
+	Action RuleActionType `json:"action"`
```

Comment:
We may want to move `Action` to the top level to match it with the [Envoy RBAC filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto) configuration. So this API can be implemented with RBAC filter.

```
  - name: envoy.filters.http.rbac
    config:
      rules:
        action: ALLOW
        policies:
          "service-admin":
            permissions:
              - any: true
            principals:
              - authenticated:
                  principal_name:
                    exact: "cluster.local/ns/default/sa/admin"
              - authenticated:
                  principal_name:
                    exact: "cluster.local/ns/default/sa/superuser"
          "product-viewer":
            permissions:
                - and_rules:
                    rules:
                      - header:
                          name: ":method"
                          string_match:
                            exact: "GET"
                      - url_path:
                          path: { prefix: "/products" }
                      - or_rules:
                          rules:
                            - destination_port: 80
                            - destination_port: 443
            principals:
              - any: true
```